### PR TITLE
fix merge for field_list_types

### PR DESCRIPTION
**Pull request by manutiedra**
This is the proper way to merge an array of arrays to allow to add values to the existing entries and to avoid the current entries to be overriden.

Now any custom made field can behave exactly as a select field.

### CODE REVIEW

**Comment by sbruner** [comment]
@manutiedra-- can you give me an example of a field that I can test. Where the values don't save with the current code, but do save with your changes?

File: includes/class-piklist-form.php [hunk 1]
```diff
@@ -315,6 +315,13 @@ public static function wp_loaded()
...
 
     // Do not allow filter to overwrite default $field_list_types
+    foreach (self::$field_list_types as $List_type => $fields)
+    {
+      if (isset($piklist_field_list_types[$List_type]))
+      {
+        self::$field_list_types[$List_type] = array_merge($piklist_field_list_types[$List_type], self::$field_list_types[$List_type]);
+      }
+    }
     self::$field_list_types = array_merge($piklist_field_list_types, self::$field_list_types);
```
Comment (JasonTheAdams) [inline includes/class-piklist-form.php]: @manutiedra How is the above code any different than line 325?

**Comment by manutiedra** [comment]
@sbruner, check https://github.com/manutiedra/piklist-autocomplete

You'll see that in the filter piklist_field_list_types an entry doesn't get added to $field_list_types['multiple_value'] without my changes. 

Note that I changed the settings from field['autocomplete'] to field['options'] in the latest code.

**Comment by sbruner** [comment]
I think I understand the issue.

Check out [this commit](https://github.com/piklist/piklist/commit/d36267c2b0d2b46f6d1eb33140cb055d99cde7da), and then change your function to this:

```
function piklist_autocomplete_field_list_types($field_list_types) {

  $field_list_types = array(
    'multiple_value' => array(
      'autocomplete'
    ));

  return $field_list_types;
}
```

If that works, I'll update the docs.

**Comment by manutiedra** [comment]
Yes @sbruner,

that fixes it.

**Comment by sbruner** [comment]
great!  Thanks for pointing out the issue.

**Pull request closed without merging**